### PR TITLE
fix(map_loader): use std::filesystem to load pcd files in pointcloud_map_loader

### PR DESCRIPTION
**Pull request by RyuYamamoto**
Signed-off-by: RyuYamamoto <ryu.yamamoto@tier4.jp>

## Description

use std::filesystem to load pcd files in pointcloud_map_loader.

## Related links

https://github.com/orgs/autowarefoundation/discussions/293

## Notes for reviewers

launch map_loader with pcd maps, check map cloud is published in topic echo and rviz.

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [x] The PR follows the [pull request guidelines].
- [x] The PR has been properly tested.
- [x] The PR has been reviewed by the code owners.

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.
- [ ] The PR is ready for merge.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

File: map/map_loader/src/pointcloud_map_loader/pointcloud_map_loader_node.cpp [hunk 1]
```diff
@@ -33,22 +33,24 @@
...
 {
 bool isPcdFile(const std::string & p)
 {
-  if (!rcutils_is_file(p.c_str())) {
+  if (fs::is_directory(p)) {
     return false;
   }
 
-  const auto ext = p.substr(p.find_last_of(".") + 1);
+  const std::string ext = fs::path(p.c_str()).extension();
```
Comment (kenji-miyake) [inline map/map_loader/src/pointcloud_map_loader/pointcloud_map_loader_node.cpp:51]: Do you need `c_str()` here?

```suggestion
  const std::string ext = fs::path(p).extension();
```

Comment (RyuYamamoto) [reply map/map_loader/src/pointcloud_map_loader/pointcloud_map_loader_node.cpp:51]: Not necessary. Sorry, I'll fix it.

File: map/map_loader/src/pointcloud_map_loader/pointcloud_map_loader_node.cpp [hunk 2]
```diff
@@ -70,21 +72,21 @@ PointCloudMapLoaderNode::PointCloudMapLoaderNode(const rclcpp::NodeOptions & opt
...
     if (isPcdFile(p)) {
       pcd_paths.push_back(p);
     }
 
-    if (rcutils_is_directory(p.c_str())) {
-      glob_t glob_buf;
-      glob((p + "/*.pcd").c_str(), 0, NULL, &glob_buf);
-      for (size_t i = 0; i < glob_buf.gl_pathc; ++i) {
-        pcd_paths.push_back(glob_buf.gl_pathv[i]);
+    if (fs::is_directory(p.c_str())) {
```
Comment (kenji-miyake) [inline map/map_loader/src/pointcloud_map_loader/pointcloud_map_loader_node.cpp:83]: Same here.
```suggestion
    if (fs::is_directory(p)) {
```

File: map/map_loader/src/pointcloud_map_loader/pointcloud_map_loader_node.cpp [hunk 3]
```diff
@@ -70,21 +72,21 @@ PointCloudMapLoaderNode::PointCloudMapLoaderNode(const rclcpp::NodeOptions & opt
...
+      for (const auto & file : fs::directory_iterator(p)) {
+        const std::string filename = file.path().c_str();
```
Comment (kenji-miyake) [inline map/map_loader/src/pointcloud_map_loader/pointcloud_map_loader_node.cpp:85]:
```suggestion
        const auto filename = file.path().string();
```

Comment (kenji-miyake) [reply map/map_loader/src/pointcloud_map_loader/pointcloud_map_loader_node.cpp:85]: Or

```suggestion
        const std::string filename = file.path().string();
```

**Review by kenji-miyake — APPROVED** [review]

File: map/map_loader/src/pointcloud_map_loader/pointcloud_map_loader_node.cpp (hunk 2 above)
Comment (RyuYamamoto) [reply map/map_loader/src/pointcloud_map_loader/pointcloud_map_loader_node.cpp]: fixed 2a560a1 .

File: map/map_loader/src/pointcloud_map_loader/pointcloud_map_loader_node.cpp (hunk 3 above)
Comment (RyuYamamoto) [reply map/map_loader/src/pointcloud_map_loader/pointcloud_map_loader_node.cpp]: fixed 2a560a1 .

**Pull request merged**